src/bot/libraries/akinator.ts

```diff
@@ -0,0 +1,348 @@
+import * as request from 'request-promise';
+
+export class Akinator {
+    private session?: Session;
+    private winner?: Character;
+
+    private certainty?: number;
+    private question?: string;
+    private step?: number;
+
+    /**
+     * Constructs a new Akinator session instance.
+     */
+    public constructor(protected region: Region = 'en') {
+
+    }
+
+    /**
+     * Starts the Akinator session. Will error if already started.
+     */
+    public async start(): Promise<Step> {
+        if (this.session) throw new Error('Akinator: Attempt to start a game which has already been started.');
+
+        // Start the session
+        let auth = await this.getAuthentication();
+        let args = await this.getSessionArgs(auth);
+
+        // Build the session object
+        this.session = {
+            sessionFrontAddr: auth.frontaddr,
+            sessionUID: auth.uid_ext_session,
+            sessionId: args.id,
+            sessionSignature: args.signature
+        };
+
+        // Set the current step and question
+        this.step = 0;
+        this.certainty = 0;
+        this.question = args.question;
+
+        // Return the step
+        return this.getStep();
+    }
+
+    /**
+     * Continues to the next step in the Akinator session.
+     */
+    public async next(answer: number): Promise<Step> {
```

Comment:
Apologies if I'm missing something, but next and previous look largely the same. The only difference I noticed is that the url in the answer response (lines 55 and 85 for next and previous respectively) has "cancel_" before "answer" in previous. You could make one method that takes a string parameter (default to "", but can pass in something like "cancel_")., change the url variable to 

`https://${this.getRequestUrl()}/ws/${<whatever your variable is called>}answer?callback=&session=${this.session.sessionId}&signature=${this.session.sessionSignature}&step=${this.step}&answer=${answer}&question_filter=`

and have next and previous just call that method. Functionally this would be basically identical, but it would be much more edit friendly as you won't have to make the same edit twice if you need to change it.

Reply:
Yeah, I had these methods separated before I realized they were the same. I figured going backwards would involve some different parameters (like `step - 1`) but noooope. Will clean it up, thanks.